src/libp2p/Libp2p.Protocols.TLS/TLS.cs

```diff
@@ -0,0 +1,408 @@
+using System.Buffers;
```

Comment:
Could you add tests that verify that TLS handshake works and data can be passed?